chia/util/streamable.py

```diff
@@ -158,16 +159,15 @@ class Streamable:
     @classmethod
     def parse_one_item(cls: Type[cls.__name__], f_type: Type, f: BinaryIO):  # type: ignore
         inner_type: Type
-        if is_type_List(f_type):
-            inner_type = get_args(f_type)[0]
-            full_list: List[inner_type] = []  # type: ignore
-            # wjb assert inner_type != get_args(List)[0]  # type: ignore
-            list_size_bytes = f.read(4)
-            assert list_size_bytes is not None and len(list_size_bytes) == 4  # Checks for EOF
-            list_size: uint32 = uint32(int.from_bytes(list_size_bytes, "big"))
-            for list_index in range(list_size):
-                full_list.append(cls.parse_one_item(inner_type, f))  # type: ignore
-            return full_list
+        if f_type is bool:
```

Comment:
One thing you might try is creating a `Dict[type, Callable]` where the callable is a function which does the parsing for that type. Then you don't have to order the checks in any certain way... it just becomes a dict look-up. It might be slower than doing several linear checks. But maybe not. (This is why python doesn't have a `switch`-type statement.)

Reply:
I guess this doesn't really work since most of the time we care if `f_type` is a *subclass* of `List` etc.